### ISSUE TITLE
Comandline program does not work

### ISSUE THREAD

**Issue by talk2sohail** [jaytaylor/go-find#1]
Hi,

I came acrosse this package while exploring linux utilities implemeted in Go. 
I am having  difficulties to execute this program with the examples given in the README.md. I am a beginner in world of Go. 
Please give me some pointers on how to get started with this? 

Thanks

**Comment by jaytaylor** [comment]
Hi,

Sorry to hear things didn't work as expected.

Are you able to share specific details on what you tried a d what the result or error messages were?

Thanks!
Jay

**Comment by talk2sohail** [comment]
Hi Jay,

Thanks for speedy reply.


I did not try much. I was just reading through readme and found out that the program did not work when I am gving command like below:


$ go-find ~/binaries/ -name 'goplay'
2023/04/29 04:22:47 lstat -name: no such file or directory

It did not work even if I give the absolute path. 

$ go-find /home/kenopasa/binaries/ -name /home/kenopasa/binaries/goplay
2023/04/29 04:24:33 lstat -name: no such file or directory


I found out with my little knowledge about Go, the "name" flag is not parsed since the parsing is stopped at first non-flag argument given in the command. I have found a workaround but I am trying to make it flexble like Unix style.

Regards,
Sohail

**Comment by talk2sohail** [comment]
Hi Jay,

I wanted to let you know I have added below feature and fixed some bugs. Please let me know if I should raise a PR. 

Feature added: 

1. Added support for multiple search path.
2. Added support to parse the flags correctly.


Regards,
Sohail

**Comment by jaytaylor** [comment]
Please do open a PR!

Can't imagine why you want a golang GNU find, lol.

**Comment by talk2sohail** [comment]
I am not sure about the changes, you would lke it or not. let me raise it anyway by EOD. Thanks!

**Comment by jaytaylor** [comment]
Ah, I see what's going on here!

So one slightly annoying thing (in this case only) is the way Go Cobra handles command-line args. It wants [flags] followed by [args] (e.g. [paths]).  I'd like to find a way to support the same argument ordering as GNU flavors of `find`, but didn't find a way to convince or even coerce Cobra into tolerating this.

Try `go-find -name /home/kenopasa/binaries/goplay /home/kenopasa/binaries`, although if `goplay` is sitting directly in the "binaries" directory, you probably want to use `-wholename` (FWIW, I believe this is the same behavior and GNU find).

Also see `go-find -h` for further usage details.

```
Usage:

        go-find [flags..] [paths...]                                                            
        Available predicate tests:

        -empty          File is empty and is either a regular file or a directory.

        -maxdepth       Descend at most levels (a non-negative integer) levels of directories below the starting-points.  -maxdepth 0 means only apply the tests and actions to the starting-points themselves.
                                                                                                        -mindepth       Do not apply any tests or actions at levels less than levels (a non-negative integer).  -mindepth 1 means process all files except the starting-points.                 
        -name           Base of file name (the path with the leading directories removed) matches shell pattern pattern.  Because the leading directories are removed, the file names considered for a match with -name will never include a slash, so `-name a/b' will never match anything (you probably need to use -path instead).

        -print0         Print the full file name on the standard output, followed by a null character (instead of the newline character).  This allows file names that contain newlines or other types of white space to be correctly interpreted by programs that process the find output.  This option corresponds to the -0 option of xargs.
                                                                                                        -regex          File name matches regular expression pattern.  This is a match on the whole path, not a search.
                                                                                                        -type           File is of type c:
                        c      character (unbuffered) special                                                           d      directory
                        p      named pipe (FIFO)                                                                        f      regular file
                        l      symbolic link; this is never true if the -L option or the -follow option is in effect, unless the symbolic link is broken.  If you want to search for symbolic links when -L is in effect, use -xtype.                                                                                   s      socket
                To search for more than one type at once, you can supply the combined list of type letters separated by a comma `,' (GNU extension).                                            
        -wholename      File name matches shell glob pattern.
```

**Comment by jaytaylor** [comment]
p.s. I hope you do submit a PR, I'll look forward to reviewing any submission you wish to make!!!

**Comment by talk2sohail** [comment]
Hi Jay,

here you go.. I have submitted a PR #2 to support the GNU-like find comand style.
Please have a look at it.

**Comment by jaytaylor** [comment]
Note: Fixed by PR #2
